src/Microsoft.OData.Client/DataServiceQueryOfT.cs

```diff
@@ -314,14 +314,27 @@ public DataServiceQuery<TElement> Expand<TTarget>(Expression<Func<TElement, TTar
         /// <returns>A new <see cref="T:Microsoft.OData.Client.DataServiceQuery`1" /> object that has the inline count option set.</returns>
         public DataServiceQuery<TElement> IncludeTotalCount()
         {
-            MethodInfo mi = typeof(DataServiceQuery<TElement>).GetMethod("IncludeTotalCount");
+            MethodInfo mi = typeof(DataServiceQuery<TElement>).GetMethods()
+                .FirstOrDefault(m => m.Name == "IncludeTotalCount" && m.GetParameters().Length == 0);
 
             return (DataServiceQuery<TElement>)this.Provider.CreateQuery<TElement>(
                 Expression.Call(
                     Expression.Convert(this.Expression, typeof(DataServiceQuery<TElement>.DataServiceOrderedQuery)),
                     mi));

```

Comment:
It's better to move all the code into the new overload method. In this old method, call the new one with the default "bool" value.

Reply:
Done